packages/ui-kit/src/helpers/chartUtils/chartUtils.ts

```diff
@@ -57,3 +66,74 @@ export const initChartJs = () => {
 
   isChartJSRegistered = true
 }
+
+export type CustomChartLabelsPluginProps = {
+  /** Sets the theme state*/
+  theme?: ThemeTokenProps
+  /** Whether the chart should show a value inside the bar */
+  showBarValues?: boolean
+  /** Sets the position of the labels */
+  labelPosition?: 'axis' | 'inside' | 'top'
+}
+
+export const getCustomChartLabelsPlugin = ({
+  theme,
+  showBarValues = false,
+  labelPosition = 'axis'
+}: CustomChartLabelsPluginProps): Plugin<ChartVariant> => {
+  return {
+    id: 'customChartLabelsPlugin',
+    afterDatasetDraw: (chart, args) => {
+      const {
+        ctx,
+        data,
+        chartArea: { left },
+        scales: { y }
+      } = chart
+
+      ctx.save()
+      ctx.textAlign = 'left'
+      ctx.textBaseline = 'middle'
+      ctx.font = `${theme?.tinyFontWeight} ${theme?.tinyFontSize} ${theme?.tinyFontFamily}`
+      ctx.fillStyle = theme?.textPrimary ?? '#ffffff'
+
+      const datasetIndex = args.index
+      const datasetMeta = chart.getDatasetMeta(datasetIndex)
+      const dataset = data.datasets[datasetIndex] as ChartDataset<ChartVariant, number[]>
+
+      if (showBarValues) {
+        dataset.data.forEach((value, index) => {
+          const barElement = datasetMeta.data[index] as BarElement
```

Comment:
It is neccessary for barChart in Leaderboard. 

this is shared custom plugin used in both Leaderboard and PieChart component.

Reply:
Can you make sure of this please? I just tried removing it in my local and everything works normal for both PieChart and Leaderboard (build, tests, etc)

We should avoid casting stuff as much as we can as it may lead to bugs in a future